openpaas-james/mailbox/encrypted/memory/src/main/java/com/linagora/openpaas/encrypted/InMemoryEncryptedMessageManager.scala

```diff
@@ -0,0 +1,87 @@
+package com.linagora.openpaas.encrypted
+
+import java.io.{ByteArrayInputStream, ByteArrayOutputStream, InputStream}
+import java.util
+import java.util.Date
+
+import com.google.common.io.ByteSource
+import com.linagora.openpaas.pgp.Encrypter
+import javax.inject.Inject
+import javax.mail.Flags
+import org.apache.james.mailbox.MessageManager.{AppendCommand, AppendResult, MailboxMetaData}
+import org.apache.james.mailbox.model.{ComposedMessageIdWithMetaData, FetchGroup, Mailbox, MailboxACL, MailboxCounters, MailboxId, MailboxPath, MessageRange, MessageResultIterator, SearchQuery}
+import org.apache.james.mailbox.{MailboxManager, MailboxSession, MessageManager, MessageUid}
+import org.reactivestreams.Publisher
+import reactor.core.scala.publisher.SFlux
+
+import scala.jdk.CollectionConverters._
+
+class InMemoryEncryptedMessageManager @Inject()(messageManager: MessageManager, keystoreManager: KeystoreManager) extends MessageManager {
+
+  override def getMessageCount(mailboxSession: MailboxSession): Long = messageManager.getMessageCount(mailboxSession)
+
+  override def getMailboxCounters(mailboxSession: MailboxSession): MailboxCounters = messageManager.getMailboxCounters(mailboxSession)
+
+  override def isWriteable(session: MailboxSession): Boolean = messageManager.isWriteable(session)
+
+  override def isModSeqPermanent(session: MailboxSession): Boolean = messageManager.isModSeqPermanent(session)
+
+  override def expunge(set: MessageRange, mailboxSession: MailboxSession): util.Iterator[MessageUid] = messageManager.expunge(set, mailboxSession)
+
+  override def search(searchQuery: SearchQuery, mailboxSession: MailboxSession): Publisher[MessageUid] = messageManager.search(searchQuery, mailboxSession)
+
+  override def delete(uids: util.List[MessageUid], mailboxSession: MailboxSession): Unit = messageManager.delete(uids, mailboxSession)
+
+  override def setFlags(flags: Flags, flagsUpdateMode: MessageManager.FlagsUpdateMode, set: MessageRange, mailboxSession: MailboxSession): util.Map[MessageUid, Flags] =
+    messageManager.setFlags(flags, flagsUpdateMode, set, mailboxSession)
+
+  override def appendMessage(msgIn: InputStream, internalDate: Date, mailboxSession: MailboxSession, isRecent: Boolean, flags: Flags): AppendResult = {
+    val keys: Seq[PublicKey] = SFlux.fromPublisher(keystoreManager.listPublicKeys(mailboxSession.getUser))
+      .collectSeq()
+      .block()
+
+    val out: ByteArrayOutputStream = new ByteArrayOutputStream
```

Comment:
Use a FIleBackedOutputStream instead, and do not forget to `reset` it once you are done with `messageManager.appendMessage`.